### ISSUE TITLE
Invalid challenge response received

### ISSUE THREAD

**Issue by tural-ali** [candy-chat/xmpp-prebind-php#18]
Installed library (using composer). So library files are untouched. XMPP server is OpenFire. 

Executing library like 

```
        $xmppPrebind = new \XmppPrebind("jabberHost", "boshUrl", 'messenger', false, true);
        $xmppPrebind->connect('user@domain.com', "password");
        $xmppPrebind->auth();
        $xmpp = $xmppPrebind->getSessionInfo();
        $result = [
            "jid" => "user@domain.com/messenger",
            "sid" => $xmpp["sid"],
            "rid" => $xmpp["rid"]
        ];
```

**Comment by cobrowserAlex** [comment]
Are you sure username and password are correct?

**Comment by tural-ali** [comment]
@cobrowserAlex yes, sure!

**Comment by tural-ali** [comment]
http://stackoverflow.com/questions/32334337/creating-persistent-connection-between-converse-js-and-openfire-server

**Comment by tural-ali** [comment]
@mweibel @cobrowserAlex anyone?

**Comment by mweibel** [comment]
@turalsprofile what do the openfire logs say?

**Comment by turalsprofile** [comment]
@mweibel  
Here is response from openfire side

```
javax.security.sasl.SaslException: DIGEST-MD5: digest response format violation. Mismatched URI: xmpp/messenger.domain.com; expecting: xmpp/domain.com
```

My Openfire is located on messenger.domain.com but accepts users like user@4pro.co. That's why my 
config is like that:
```
        $xmppPrebind = new \XmppPrebind('messenger.domain.com', http://messenger.domain.com:7070/http-bind/, 'website', false, true);
```
What am I doing wrong?

**Comment by turalsprofile** [comment]
@mweibel 

I tried also

```
        $xmppPrebind = new \XmppPrebind('domain.com', http://messenger.domain.com:7070/http-bind/, 'website', false, true);
```

In this case php side keeps loading about 1 minutes that gives 504 gateway error. Openfire debug shows like below:


```
2015.09.21 02:13:10 org.jivesoftware.openfire.http.HttpBindServlet - Data can be written to [188.226.254.15] 
2015.09.21 02:13:10 org.jivesoftware.openfire.http.HttpBindServlet - Data is available to be read from [188.226.254.15] 
2015.09.21 02:13:10 org.jivesoftware.openfire.http.HttpBindServlet - All data has been read from [188.226.254.15] 
2015.09.21 02:13:10 org.jivesoftware.openfire.http.HttpBindServlet - Data can be written to [188.226.254.15] 
2015.09.21 02:13:10 org.jivesoftware.openfire.http.HttpSession - complete event org.eclipse.jetty.server.AsyncContextEvent@6d2b815f 
2015.09.21 02:13:10 org.jivesoftware.openfire.http.HttpBindServlet - Data is available to be read from [188.226.254.15] 
2015.09.21 02:13:10 org.jivesoftware.openfire.http.HttpBindServlet - All data has been read from [188.226.254.15] 
2015.09.21 02:13:10 org.jivesoftware.openfire.auth.AuthorizationManager - AuthorizationManager: Trying Default Mapping.map(tural) 
2015.09.21 02:13:10 org.jivesoftware.openfire.auth.DefaultAuthorizationMapping - DefaultAuthorizationMapping: No realm found 
2015.09.21 02:13:10 org.jivesoftware.openfire.auth.AuthorizationManager - AuthorizationManager: Trying Default Policy.authorize(tural , tural) 
2015.09.21 02:13:10 org.jivesoftware.openfire.auth.DefaultAuthorizationPolicy - DefaultAuthorizationPolicy: Checking authenID realm 
2015.09.21 02:13:10 org.jivesoftware.openfire.http.HttpBindServlet - Data can be written to [188.226.254.15] 
2015.09.21 02:13:10 org.jivesoftware.openfire.http.HttpSession - complete event org.eclipse.jetty.server.AsyncContextEvent@527c9cda 
2015.09.21 02:13:10 org.jivesoftware.openfire.http.HttpBindServlet - Data is available to be read from [188.226.254.15] 
2015.09.21 02:13:10 org.jivesoftware.openfire.http.HttpBindServlet - All data has been read from [188.226.254.15] 
2015.09.21 02:13:12 org.jivesoftware.util.log.util.CommonsLogFactory - Closing statement 7c289d0f (belonging to connection 62) automatically 
2015.09.21 02:13:16 org.jivesoftware.openfire.http.HttpSessionManager - Session was last active 0 ms ago: 4pro.co/495dab84 
2015.09.21 02:13:16 org.jivesoftware.openfire.http.HttpSessionManager - Session was last active -1 ms ago: 4pro.co/3fda2187 
2015.09.21 02:13:16 org.jivesoftware.openfire.http.HttpSessionManager - Session was last active -1 ms ago: 4pro.co/54996245 
2015.09.21 02:13:18 org.jivesoftware.openfire.http.HttpSession - timeout event org.eclipse.jetty.server.AsyncContextEvent@3ab27f62 
2015.09.21 02:13:18 org.jivesoftware.openfire.http.HttpSession - complete event org.eclipse.jetty.server.AsyncContextEvent@3ab27f62 
2015.09.21 02:13:20 org.jivesoftware.openfire.http.HttpSession - timeout event org.eclipse.jetty.server.AsyncContextEvent@568aaca2 
2015.09.21 02:13:20 org.jivesoftware.openfire.http.HttpSession - complete event org.eclipse.jetty.server.AsyncContextEvent@568aaca2 
2015.09.21 02:13:34 org.jivesoftware.openfire.reporting.stats.StatsEngine - Stat: muc_traffic. Last sample: 1442815920. New sample: 1442815980 
2015.09.21 02:13:34 org.jivesoftware.openfire.reporting.stats.StatsEngine - Stat: server_sessions. Last sample: 1442815920. New sample: 1442815980 
2015.09.21 02:13:37 org.apache.mina.filter.ssl.SslFilter - Session Server[228](SSL): Message received : HeapBuffer[pos=0 lim=74 cap=128: 17 03 01 00 20 68 0B 8E 23 53 03 9B 5E BC 37 42...] 
2015.09.21 02:13:37 org.apache.mina.filter.ssl.SslHandler - Session Server[228](SSL) Processing the received message 
2015.09.21 02:13:37 org.apache.mina.filter.ssl.SslFilter - Session Server[228](SSL): Processing the SSL Data 
2015.09.21 02:13:37 org.apache.mina.filter.executor.OrderedThreadPoolExecutor - Adding event MESSAGE_RECEIVED to session 228 
Queue : [MESSAGE_RECEIVED, ] 

2015.09.21 02:13:37 org.apache.mina.filter.codec.ProtocolCodecFilter - Processing a MESSAGE_RECEIVED for session 228 
2015.09.21 02:13:46 org.jivesoftware.openfire.http.HttpSessionManager - Session was last active 26140 ms ago: 4pro.co/495dab84 
2015.09.21 02:13:46 org.jivesoftware.openfire.http.HttpSessionManager - Session was last active -1 ms ago: 4pro.co/3fda2187 
2015.09.21 02:13:46 org.jivesoftware.openfire.http.HttpSessionManager - Session was last active 27423 ms ago: 4pro.co/54996245 
2015.09.21 02:14:10 org.jivesoftware.openfire.http.HttpSession - timeout event org.eclipse.jetty.server.AsyncContextEvent@71d8b8a4 
2015.09.21 02:14:10 org.jivesoftware.openfire.http.HttpSession - complete event org.eclipse.jetty.server.AsyncContextEvent@71d8b8a4 
2015.09.21 02:14:16 org.jivesoftware.openfire.http.HttpSessionManager - Session was last active 56141 ms ago: 4pro.co/495dab84 
2015.09.21 02:14:16 org.jivesoftware.openfire.http.HttpSessionManager - Session was last active 5693 ms ago: 4pro.co/3fda2187 
2015.09.21 02:14:16 org.jivesoftware.openfire.http.HttpSessionManager - Session was last active 57424 ms ago: 4pro.co/54996245 
2015.09.21 02:14:34 org.jivesoftware.openfire.reporting.stats.StatsEngine - Stat: muc_traffic. Last sample: 1442815980. New sample: 1442816040 
2015.09.21 02:14:34 org.jivesoftware.openfire.reporting.stats.StatsEngine - Stat: server_sessions. Last sample: 1442815980. New sample: 1442816040 
2015.09.21 02:14:37 org.apache.mina.filter.ssl.SslFilter - Session Server[228](SSL): Message received : HeapBuffer[pos=0 lim=74 cap=128: 17 03 01 00 20 6C BB 14 90 A7 CC C3 AF 4C FE 11...] 
2015.09.21 02:14:37 org.apache.mina.filter.ssl.SslHandler - Session Server[228](SSL) Processing the received message 
2015.09.21 02:14:37 org.apache.mina.filter.ssl.SslFilter - Session Server[228](SSL): Processing the SSL Data 
2015.09.21 02:14:37 org.apache.mina.filter.executor.OrderedThreadPoolExecutor - Adding event MESSAGE_RECEIVED to session 228 
Queue : [MESSAGE_RECEIVED, ] 

2015.09.21 02:14:37 org.apache.mina.filter.codec.ProtocolCodecFilter - Processing a MESSAGE_RECEIVED for session 228 
2015.09.21 02:14:46 org.jivesoftware.openfire.http.HttpSessionManager - Session was last active 35693 ms ago: 4pro.co/3fda2187 

```

**Comment by turalsprofile** [comment]
@mweibel why do you think it's happening?
